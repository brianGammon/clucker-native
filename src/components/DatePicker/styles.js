import { StyleSheet } from 'react-native';
import variables from 'native-base/dist/src/theme/variables/platform';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: variables.inputBorderColor,
  },
  datePickerField: {
    flex: 1,
  },
  label: {
    fontWeight: 'bold',
    color: variables.inputColorPlaceholder,
  },
  clearDateIcon: {
    color: variables.textColor,
  },
  placeHolderTextStyle: {
    color: variables.inputColorPlaceholder,
  },
});
