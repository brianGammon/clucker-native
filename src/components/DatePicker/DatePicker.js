/* @flow */
import React from 'react';
import moment from 'moment';
import {
  DatePicker as NBDatePicker, View, Button, Icon,
} from 'native-base';
import CommonLabel from '../CommonLabel';
import { nowAsMoment } from '../../utils/dateHelper';
import styles from './styles';

type Props = {
  clearable: boolean,
  label: string,
  value: Date | null,
  minimumDate: Date | null,
  maximumDate: Date | null,
  onDateChange: (date: string) => void,
};

class DatePicker extends React.Component<Props> {
  datePickerRef: { current: null | NBDatePicker } = React.createRef();

  handleClearDate = () => {
    if (this.datePickerRef.current) {
      this.datePickerRef.current.setState({ chosenDate: undefined }, () => {
        if (this.datePickerRef.current) {
          this.datePickerRef.current.props.onDateChange(null);
        }
      });
    }
  };

  handleChangeDate = (date: Date | null) => {
    const { onDateChange } = this.props;
    let dateAsString = '';
    if (date) {
      dateAsString = moment(date).format('YYYY-MM-DD');
    }
    onDateChange(dateAsString);
  };

  render() {
    const {
      clearable, value, maximumDate, minimumDate, label,
    } = this.props;
    const valueAsDate = value && value !== '' ? new Date(moment(value).valueOf()) : null;
    return (
      <View style={styles.container}>
        <CommonLabel text={`${label || 'Date'}:`} />
        <View style={styles.datePickerField}>
          <NBDatePicker
            ref={this.datePickerRef}
            defaultDate={valueAsDate}
            minimumDate={
              minimumDate
              || new Date(
                nowAsMoment()
                  .subtract(2, 'year')
                  .valueOf(),
              )
            }
            maximumDate={maximumDate}
            locale="en"
            modalTransparent
            animationType="fade"
            androidMode="default"
            placeHolderTextStyle={styles.placeHolderTextStyle}
            onDateChange={this.handleChangeDate}
            formatChosenDate={date => moment(date).format('MMM DD, YYYY')}
          />
        </View>
        {valueAsDate
          && clearable && (
            <Button transparent onPress={this.handleClearDate}>
              <Icon style={styles.clearDateIcon} name="close-circle" />
            </Button>
        )}
      </View>
    );
  }
}

export default DatePicker;
