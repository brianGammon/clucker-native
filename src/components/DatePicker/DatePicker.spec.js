import React from 'react';
import DatePicker from '.';

describe('DatePicker', () => {
  const props = {
    clearable: true,
    value: new Date('8/31/2018'),
    maximumDate: new Date('12/31/2018'),
    onDateChange: () => {},
  };
  test('renders with all props supplied', () => {
    const wrapper = shallow(<DatePicker {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
