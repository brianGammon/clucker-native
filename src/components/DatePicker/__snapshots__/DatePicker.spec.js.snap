// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders with all props supplied 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#D9D5DC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <CommonLabel
    style={null}
    text="Date:"
  />
  <Styled(ViewNB)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <DatePicker
      androidMode="default"
      animationType="fade"
      defaultDate={2018-08-31T04:00:00.000Z}
      formatChosenDate={[Function]}
      locale="en"
      maximumDate={2018-12-31T05:00:00.000Z}
      minimumDate={2016-09-16T00:00:00.000Z}
      modalTransparent={true}
      onDateChange={[Function]}
      placeHolderTextStyle={
        Object {
          "color": "#575757",
        }
      }
    />
  </Styled(ViewNB)>
  <Styled(Button)
    onPress={[Function]}
    transparent={true}
  >
    <Styled(Icon)
      name="close-circle"
      style={
        Object {
          "color": "#000",
        }
      }
    />
  </Styled(Button)>
</Styled(ViewNB)>
`;
